src/sql/workbench/browser/modal/modal.ts

```diff
@@ -391,14 +391,16 @@ export abstract class Modal extends Disposable implements IThemable {
 	/**
 	 * Hides the modal and removes key listeners
 	 */
-	protected hide(reason?: string) {
+	protected hide(reason?: string, currentPageTitle?: string) {
 		this._modalShowingContext.get()!.pop();
 		this._bodyContainer!.remove();
 		this.disposableStore.clear();
 		this._telemetryService.createActionEvent(TelemetryKeys.TelemetryView.Shell, TelemetryKeys.ModalDialogClosed)
 			.withAdditionalProperties({
 				name: this._name,
-				reason: reason
+				reason: reason,
+				wizard: this._title,
```

Comment:
The name field is supposed to be the identifier for the dialog - not the title. We should avoid using displayed titles for telemetry since that may contain PII (such as if we have a wizard/page with the name of the target database in it).

The name field is specifically to be used for telemetry so should be a safe value that's the same for all instances of the wizard/page.

Reply:
So I'd suggest adding an optional name field to the page - and then if that isn't provided by the extension just default to using the index of the page instead.